core/embed/rust/src/ui/model_tr/layout.rs

```diff
@@ -638,131 +639,112 @@ extern "C" fn new_confirm_output_amount(n_args: usize, args: *const Obj, kwargs:
     unsafe { util::try_with_args_and_kwargs(n_args, args, kwargs, block) }
 }
 
-extern "C" fn new_confirm_total(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj {
+extern "C" fn new_confirm_summary(n_args: usize, args: *const Obj, kwargs: *mut Map) -> Obj {
     let block = |_args: &[Obj], kwargs: &Map| {
-        let total_amount: TString = kwargs.get(Qstr::MP_QSTR_total_amount)?.try_into()?;
-        let fee_amount: TString = kwargs.get(Qstr::MP_QSTR_fee_amount)?.try_into()?;
-        let fee_rate_amount: Option<TString> = kwargs
-            .get(Qstr::MP_QSTR_fee_rate_amount)?
-            .try_into_option()?;
-        let account_label: Option<TString> =
-            kwargs.get(Qstr::MP_QSTR_account_label)?.try_into_option()?;
-        let total_label: TString = kwargs.get(Qstr::MP_QSTR_total_label)?.try_into()?;
+        let amount: TString = kwargs.get(Qstr::MP_QSTR_amount)?.try_into()?;
+        let amount_label: TString = kwargs.get(Qstr::MP_QSTR_amount_label)?.try_into()?;
+        let fee: TString = kwargs.get(Qstr::MP_QSTR_fee)?.try_into()?;
         let fee_label: TString = kwargs.get(Qstr::MP_QSTR_fee_label)?.try_into()?;
+        let _title: Option<TString> = kwargs
+            .get(Qstr::MP_QSTR_title)
+            .unwrap_or_else(|_| Obj::const_none())
+            .try_into_option()?;
+        let fee_items: Option<Obj> = kwargs
+            .get(Qstr::MP_QSTR_fee_items)
+            .unwrap_or_else(|_| Obj::const_none())
+            .try_into_option()?;
+        let account_items: Option<Obj> = kwargs
+            .get(Qstr::MP_QSTR_account_items)
+            .unwrap_or_else(|_| Obj::const_none())
+            .try_into_option()?;
+        let extra_items: Option<Obj> = kwargs
+            .get(Qstr::MP_QSTR_extra_items)
+            .unwrap_or_else(|_| Obj::const_none())
+            .try_into_option()?;
+        let cancel_arrow: bool = kwargs.get_or(Qstr::MP_QSTR_cancel_arrow, false)?;
```

Comment:
I think that on model R it should be `verb_cancel = "^"` versus `verb_cancel = ""` to mark the "up arrow" versus the "x" icon. I've seen that used in other places. Introducing the concept of "cancel_arrow" seems wrong, not just because it's unclear what it means, but also because it is not portable across models.

Reply:
Both `TT` and `TR` either uses "x" or "^" as the cancel button so that you can use "^" if the summary is a part of uPy flow.

`TT` uses `cancel_arrow` now and it used it before.
`TR` used `cancel_cross` before in `altcoin_tx_summary` which was basically the opposite of what `cancel_arrow` does. So I unified it. 
`Lincoln` will probably need it as well given current designs.
`Mercury` does not need it but maybe it will if the designs change in the future based on Lincoln.

Having `verb_cancel` there might make more sense. Let me try.

Reply:
made a mistake for `TR`, fixed here: 61448aad34b21198d00d80507fc3321b923be27a